File: lab4/backend/products/soa-products-rest-adapter/src/main/kotlin/com/soa/products/rest/adapter/wsdl/PriceOperationTo.java

package com.soa.products.rest.adapter.wsdl;

import jakarta.xml.bind.annotation.XmlEnum;
import jakarta.xml.bind.annotation.XmlType;


/**
 * 
 * 
 * <p>Java class for priceOperationTo</p>.
 * 
 * <p>The following schema fragment specifies the expected content contained within this class.</p>
 * <pre>{@code
 * <simpleType name="priceOperationTo">
 *   <restriction base="{http://www.w3.org/2001/XMLSchema}string">
 *     <enumeration value="INCREASE"/>
 *     <enumeration value="DECREASE"/>
 *   </restriction>
 * </simpleType>
 * }</pre>
 * 
 */
@XmlType(name = "priceOperationTo")
@XmlEnum
public enum PriceOperationTo {

    INCREASE,
    DECREASE;

    public String value() {
        return name();
    }

    public static PriceOperationTo fromValue(String v) {
        return valueOf(v);
    }

}

File: lab4/backend/products/soa-products/src/main/kotlin/com/soa/products/dto/PriceOperationTo.kt
package com.soa.products.dto

enum class PriceOperationTo(val value: String) {

    INCREASE("INCREASE"),
    DECREASE("DECREASE");

    companion object {
        fun forValue(value: String): generated.soa.products.dto.PriceOperationTo {
            return generated.soa.products.dto.PriceOperationTo.values().first{ it.value == value}
        }
    }
}
